components/_impl/workers/subprocess_worker.py

```diff
@@ -49,12 +49,18 @@ def args(self) -> typing.List[str]:
     """
 
     _working_dir: str
+    _user_save_output: bool = False
     _alive: bool = False
     _bootstrap_timeout: int = 10  # seconds
 
-    def __init__(self, timeout: typing.Optional[float] = None, extra_env: typing.Optional[typing.Dict[str, str]]=None) -> None:
+    def __init__(self, timeout: typing.Optional[float] = None,
+                 extra_env: typing.Optional[typing.Dict[str, str]]=None,
```

Comment:
to improve readability, we could use `from typing import Dict, Optional`, and use those directly. Also, seems inconsistently spacing before and after `=` in line 56-58.
```
self, timeout: Optional[float]=None,
extra_env: Optional[Dict[str, str]]=None,
save_output_dir: pathlib.Path=None) -> None:
```

Reply:
Addressed the formatting issue